### PR TITLE
Update items on form  

### PR DESCRIPTION
**Pull request by shannonmoranetz**
Added redux state for: 

- Current note (object)
- Show pop-up (boolean)
- Updated note (object) 

Form now opens/closes when clicking submit button / clicking 'Add Note' button. 
Form uses put method functionality from backend to update the current note. 
Backend and redux state kept in sync.

### CODE REVIEW

**Comment by JustinD85** [comment]
Co-authored-by: Justin Duncan<justin@baal-software.com>

**Comment by JustinD85** [comment]
b

File: src/containers/App/App.js [hunk 1]
```diff
@@ -3,39 +3,46 @@ import { NoteArea, CreateNote } from "../";
...
-        <CreateNote canRender={showPopup} />
+        {this.props.shouldDisplay && <CreateNote />}
         <NoteArea />
       </div>
     );
   }
 }
 
+export const mapStateToProps = state => ({
+  shouldDisplay: state.shouldDisplay
```
Comment (dylhof) [inline src/containers/App/App.js]: So is ‘shouldDisplay’ the property in state controlled by the action ‘showPopUp’?

File: src/reducers/notesReducer.js [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+import { ADD_NEW_NOTE, ADD_ALL_NOTES, UPDATE_NOTE } from "../actions";
 
 export function notesReducer(state = [], action) {
   switch (action.type) {
     case ADD_NEW_NOTE:
       return [...state, action.newNote];
     case ADD_ALL_NOTES:
       return action.notes
+    case UPDATE_NOTE:
+      console.log(action)
```
Comment (dylhof) [inline src/reducers/notesReducer.js]: Let’s make sure to remove this co sole log 🙂

File: src/containers/App/App.js (hunk 1 above)
Comment (dylhof) [reply src/containers/App/App.js]: Okay! I see this later on.

File: src/thunks/index.test.js [hunk 3]
```diff
@@ -1,10 +1,7 @@
 import { getNotes, postNotes } from './index';
 import * as actions from '../actions';
```
Comment (dylhof) [inline src/thunks/index.test.js]: Thanks for removing the unused imports and variables!! 😅

File: src/containers/CreateNote/CreateNote.js [hunk 4]
```diff
@@ -67,12 +66,17 @@ class CreateNote extends React.Component {
...
     event.preventDefault();
-    // const { addNewNote } = this.props;
     const { title, noteItems } = this.state;
-    this.props.postNote({ title, noteItems, id: uuid() });
-    // addNewNote({ title, noteItems });
+    const { id: noteId } = this.props.currentNote;
+    if (noteId) {
+      this.props.putNote({ title, noteItems, id: noteId});
+      this.props.setCurrentNote({});
+      this.props.showPopUp(false);
```
Comment (dylhof) [inline src/containers/CreateNote/CreateNote.js]: This is really good! I like the way you found to reuse this component! Great work! 😄

**Pull request merged**